src/middlewared/middlewared/etc_files/syslogd.py

```diff
@@ -0,0 +1,178 @@
+from datetime import datetime
+import logging
+import os
+import shlex
+import shutil
+import subprocess
+import textwrap
+
+logger = logging.getLogger(__name__)
+
+
+def generate_syslog_conf(middleware):
+    # In 12.0 we can have /conf/base/usr__local__etc as the base point for /usr/local/etc/tmpfs
+    # (see commit 704f1eb60f438171690d79bfdf17e95044cc6bb2)
+    if os.path.isdir("/conf/base/usr__local__etc"):
+        shutil.copy("/conf/base/usr__local__etc/syslog-ng.conf.freenas",
+                    "/usr/local/etc/syslog-ng.conf")
```

Comment:
@kmoore134 it is `"/usr/local/etc/syslog-ng.conf"`but other parts of this file are not changed, is this on purpose?

Reply:
@themylogin can you elaborate on your question? I'm not sure I understand it but I might be able to help answer.

Reply:
@yocalebo as 704f1eb60f438171690d79bfdf17e95044cc6bb2 commit message says, in 12.0 we should use `/conf/base/usr__local__etc/syslog-ng.conf.freenas` as syslog-ng template instead of `/conf/base/etc/local/syslog-ng.conf.freenas`, but we also put it to `/usr/local/etc/syslog-ng.conf` instead of `/etc/local/syslog-ng.conf`, and code below only patches the latter. Which config file is really used on 12.0?

Reply:
@themylogin /usr/local/etc and /etc/local are symlinks to the same tmpfs partition.